cfgmgr/coppmgr.cpp

```diff
@@ -0,0 +1,859 @@
+            }
+            m_coppTrapConfMap[key].trap_group = trap_group;
+            m_coppTrapConfMap[key].trap_ids = trap_ids;
+            setCoppTrapStateOk(key);
+        }
+        else if (op == DEL_COMMAND)
+        {
+            removeTrapIdsFromTrapGroup(m_coppTrapConfMap[key].trap_group,
+                                       m_coppTrapConfMap[key].trap_ids);
+            fvs.clear();
+            trap_ids.clear();
+            if (!m_coppTrapConfMap[key].trap_group.empty())
+            {
+                getTrapGroupTrapIds(m_coppTrapConfMap[key].trap_group, trap_ids);
+                FieldValueTuple fv(COPP_TRAP_ID_LIST_FIELD, trap_ids);
+                fvs.push_back(fv);
+                if (!checkIfTrapGroupFeaturePending(m_coppTrapConfMap[key].trap_group))
+                {
+                    m_appCoppTable.set(m_coppTrapConfMap[key].trap_group, fvs);
+                    setCoppGroupStateOk(m_coppTrapConfMap[key].trap_group);
+                }
+            }
+            m_coppTrapConfMap.erase(key);
+            delCoppTrapStateOk(key);
+
+            /* If the COPP trap was part of init config, it needs to be recreated
+             * with field values from init. The configuration delete should just clear
+             * the externally applied user configuration
+             */
+            if (m_coppTrapInitCfg.find(key) != m_coppTrapInitCfg.end())
+            {
+                auto fvs = m_coppTrapInitCfg[key];
+                for (auto i: fvs)
+                {
+                    if (fvField(i) == COPP_TRAP_GROUP_FIELD)
+                    {
+                        trap_group = fvValue(i);
+                    }
+                    else if (fvField(i) == COPP_TRAP_ID_LIST_FIELD)
+                    {
+                        trap_ids = fvValue(i);
+                    }
+                }
+                vector<FieldValueTuple> g_fvs;
+                string trap_group_trap_ids;
+                addTrapIdsToTrapGroup(trap_group, trap_ids);
+                getTrapGroupTrapIds(trap_group, trap_group_trap_ids);
+                FieldValueTuple fv1(COPP_TRAP_ID_LIST_FIELD, trap_group_trap_ids);
+                g_fvs.push_back(fv1);
+                if (!checkIfTrapGroupFeaturePending(trap_group))
+                {
+                    m_appCoppTable.set(trap_group, g_fvs);
+                    setCoppGroupStateOk(trap_group);
+                }
+                m_coppTrapConfMap[key].trap_group = trap_group;
+                m_coppTrapConfMap[key].trap_ids = trap_ids;
+                setCoppTrapStateOk(key);
+            }
+        }
+        it = consumer.m_toSync.erase(it);
+    }
+}
+
+/* This API is used to fetch only the modified configurations. During warmboot
+ * when certain fields in APP table are not present in new init config file
+ * the APP table needs to be removed and recreated
+ */
+void CoppMgr::coppGroupGetModifiedFvs(string key, vector<FieldValueTuple> &trap_group_fvs, 
```

Comment:
Why do we need this with db_migrator changes to remove the CoPP tables from APP_DB during warmboot?

Reply:
This API is not just used in warm reboot but also in general set. When a set is done in config DB, we do not get just the modified fields but rather all the key value pairs. Thus we would be doing redundant set operations into app db. This will also create issues when the ASIC DB has create only fields and the set to APP_DB triggers set operation. This API would just set the modified key value pairs into app DB and thus preventing such issues